### ISSUE TITLE
Promotion: Rule value disappears on rule type change

### ISSUE THREAD

**Issue by igormukhingmailcom** [Sylius/Sylius#10413]
**Sylius version affected**: 1.3 (at least, but I guess any)

**Description**  
When we change rule type at promotion add/change admin form, configuration values(-s) no longer displayed.

**Steps to reproduce**  
- Open any promotion http://localhost:8000/admin/promotions/1/edit
- Remember what Rule Type was and what configuration value(-s) was (lets define them as **initial**)
- Change Rule Type to any other
- Change Rule Type back to **initial**
- Configuration value(-s) not restoring (configuration field(-s) is blank) <---- ISSUE

**Possible Solution**  
Not sure. Some JS handler should be called on Rule Type change I guess, not only on page load.

**Comment by Zales0123** [comment]
@kulczy do you think it can be somehow resolved on a js level?

**Comment by igormukhingmailcom** [comment]
Yeah, initial value should be stored in js variable I guess and once user
switch back to initial state - that value should be restored. Wdyt?

пн, 10 июн. 2019 г., 12:26 Mateusz Zalewski <notifications@github.com>:

**Comment by kulczy** [comment]
Hi, @igormukhingmailcom thanks for reporting. I have no doubt that this is useful and we will probably think about it in the future, but we currently have things with a higher priority
